src/Language/Haskell/LSP/VFS.hs

```diff
@@ -62,35 +67,49 @@ data VirtualFile =
   VirtualFile {
       _version :: Int
     , _text    :: Rope
-    , _tmp_file :: Maybe FilePath
     } deriving (Show)
 
-type VFS = Map.Map J.NormalizedUri VirtualFile
+type VFSMap = Map.Map J.NormalizedUri VirtualFile
+
+data VFS = VFS { vfsMap :: Map.Map J.NormalizedUri VirtualFile
+               , vfsTempDir :: FilePath -- ^ This is where all the temporary files will be written to
+               } deriving Show
```

Comment:
I think we should move the `revMap` in here too.

Reply:
Done